cobbler/cobbler_collections/collection.py

```diff
@@ -235,9 +235,18 @@ def rename(self, ref: item_base.Item, newname, with_sync: bool = True, with_trig
         if newname == ref.name:
             return
 
-        # Save the old name and rename object
+        # Save the old name
         oldname = ref.name
+        # Reserve the new name
+        self.listing[newname] = None
+        # Delete the old item
+        self.collection_mgr.serialize_delete_one_item(ref)
+        self.listing.pop(oldname)
+        # Change the name of the object
         ref.name = newname
+        # Save just this item
+        self.collection_mgr.serialize_one_item(ref)
+        self.listing[newname] = ref
```

Comment:
I'm not really happy about mixing file I/O (which might go wrong for a number of reasons) with the manipulation of the `self.listing` dictionary, but I don't see a better way to handle this issue right now.